specs/experimental/op-contracts-manager.md

```diff
@@ -280,36 +271,45 @@ The `addGameType` method is used to orchestrate the actions required to add a ne
 or more chains.
 
 ```solidity
-struct PermissionlessGameConfig {
-  bytes32 absolutePrestate;
+struct AddGameInput {
+    string saltMixer;
+    ISystemConfig systemConfig;
+    IProxyAdmin proxyAdmin;
+    IDelayedWETH delayedWETH;
+    GameType disputeGameType;
+    Claim disputeAbsolutePrestate;
+    uint256 disputeMaxGameDepth;
+    uint256 disputeSplitDepth;
+    Duration disputeClockExtension;
+    Duration disputeMaxClockDuration;
+    uint256 initialBond;
+    IBigStepper vm;
+    bool permissioned;
+}
+
+struct AddGameOutput {
+    IDelayedWETH delayedWETH;
+    IFaultDisputeGame faultDisputeGame;
 }
 
-function addGameType(ISystemConfig[] _systemConfigs, PermissionlessGameConfig[] _newGames) public;
+/// @notice addGameType deploys a new dispute game and links it to the DisputeGameFactory. The inputted _gameConfigs
+/// must be added in ascending GameType order.
+function addGameType(AddGameInput[] memory _gameConfigs) external returns (AddGameOutput[] memory)
 ```
 
 ### Implementation
 
 The high level logic of the `addGameType` method is as follows (for each chain):
 
-1. Deploy and initialize new `DelayedWethProxy` for the new game type, reusing the existing implementation
-1. Deploy a new `FaultDisputeGame` contract. The source of the constructor args is indicated below this list,
-   the value which is not available onchain is the `absolutePrestate`.
-1. Calls `upgrade()` on the `AnchorStateRegistry` to set the new game type to
-   add a new entry to the `anchors` mapping. The `upgrade()` method should
-   revert if it would overwrite an existing entry.
-1. Read the `DisputeGameFactory` address from the `SystemConfig`.
-1. Call `DisputeGameFactory.setImplementation()` to register the new game.
+1. Deploy and initialize new `DelayedWethProxy` for the new game type, if one hasn't already been specified.
+2. Deploy a new `FaultDisputeGame` contract. The constructor args must be provided as arguments, except as per the
+   table below.
```

Comment:
```suggestion
2. Deploy a new dispute game contract, based on the specified game type. The constructor args must be provided as arguments, unless otherwise specified in
   the table below.
```